docs/content/envoy-authorization.md

```diff
@@ -0,0 +1,334 @@
+
+## Prerequisites
+
+This tutorial requires Kubernetes 1.14 or later. To run the tutorial locally, we recommend using [minikube](https://kubernetes.io/docs/getting-started-guides/minikube) in version `v1.0+` with Kubernetes 1.14 (which is the default).
+
+## Steps
+
+### 1. Start Minikube
+
+```bash
+minikube start
+```
+
+### 2. Create ConfigMaps containing configuration for Envoy and OPA
+
+The Envoy configuration below defines an external authorization filter `envoy.ext_authz` for a gRPC authorization server.
+
+Save the configuration as **envoy.yaml**:
+
+```yaml
+static_resources:
+  listeners:
+    - address:
+        socket_address:
+          address: 0.0.0.0
+          port_value: 8000
+      filter_chains:
+        - filters:
+            - name: envoy.http_connection_manager
+              typed_config:
+                "@type": type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager
+                codec_type: auto
+                stat_prefix: ingress_http
+                route_config:
+                  name: local_route
+                  virtual_hosts:
+                    - name: backend
+                      domains:
+                        - "*"
+                      routes:
+                        - match:
+                            prefix: "/"
+                          route:
+                            cluster: service
+                http_filters:
+                  - name: envoy.ext_authz
+                    config:
+                      with_request_body:
+                        max_request_bytes: 8192
+                        allow_partial_message: true
+                      failure_mode_allow: false
+                      grpc_service:
+                        google_grpc:
+                          target_uri: 127.0.0.1:9191
+                          stat_prefix: ext_authz
+                        timeout: 0.5s
+                  - name: envoy.router
+                    typed_config: {}
+  clusters:
+    - name: service
+      connect_timeout: 0.25s
+      type: strict_dns
+      lb_policy: round_robin
+      load_assignment:
+        cluster_name: service
+        endpoints:
+          - lb_endpoints:
+              - endpoint:
+                  address:
+                    socket_address:
+                      address: 127.0.0.1
+                      port_value: 8080
+admin:
+  access_log_path: "/dev/null"
+  address:
+    socket_address:
+      address: 0.0.0.0
+      port_value: 8001
+```
+
+Create the ConfigMap:
+
+```bash
+kubectl create configmap proxy-config --from-file envoy.yaml
+```
+
+The OPA configuration below sets the listening address of Envoy External Authorization gRPC server and specifies the name of the policy decision to query. More information on the configuration options can be found [here](https://github.com/open-policy-agent/opa-istio-plugin#configuration).
+
+Save the configuration as **opa.yaml**:
+
+```yaml
```

Comment:
I believe we can set all of these options on the command line for OPA. For example:

```
--set=plugins.envoy_ext_authz_grpc.addr=:9191
```

Let's switch to that and avoid the need for another configmap.

Reply:
Done.